privacyidea/static/components/token/controllers/tokenDetailController.js

```diff
@@ -190,9 +190,42 @@ myApp.controller("tokenDetailController", function ($scope,
         }, $scope.get);
     };
 
-    $scope.delete = function () {
-        TokenFactory.delete($scope.tokenSerial, $scope.return_to);
+    $scope.deleteTokenAsk = function() {
+            $('#dialogTokenDelete').modal();
```

Comment:
I think this line needs to be removed. My suspicion is, that is would cause the modal to always show, even if the token is not a hardware token.

Reply:
i tested it in venv and it does work as intended.